danesfield/core/tasks/__init__.py

```diff
@@ -13,12 +13,38 @@
 from django.core.files.uploadedfile import SimpleUploadedFile
 import requests
 from rgd.models.common import ChecksumFile
+from rgd_3d.models import PointCloud
+from rgd_imagery.models.base import Image, ImageSet
 
 # Prevent circular import
 from danesfield.core.models import Dataset, DatasetRun
 
 logger = get_task_logger(__name__)
 
+RGD_IMAGERY_EXTENSIONS = ('.tif', '.png')
+RGD_3D_EXTENSIONS = ('.ply', '.obj')
+
+
+def _ingest_checksum_files(files: List[ChecksumFile]):
+    images: List[Image] = []
+    meshes: List[PointCloud] = []
+    for checksum_file in files:
+        extension: str = Path(checksum_file.name).suffix
+
+        if not extension:
+            break
+
+        if extension in RGD_IMAGERY_EXTENSIONS:
+            images.append(Image(file=checksum_file))
+        elif extension in RGD_3D_EXTENSIONS:
+            meshes.append(PointCloud(file=checksum_file))
+
+    if images:
+        ImageSet.objects.create().images.set(Image.objects.bulk_create(images))
```

Comment:
What's the reason for creating an `ImageSet` from all images in the output files? Is it just for organization? I'm not necessarily against them being grouped into an image set, but I don't exactly see how we'd make use of it.

Reply:
The only reason is so that they are viewable with the RGD SSR pages, since to my knowledge those require `Images` to be part of an `ImageSet` to be displayed.